handlers/infobip/handler.go

```diff
@@ -160,51 +223,193 @@ func (h *handler) receiveMessage(ctx context.Context, channel courier.Channel, w
 	return handlers.WriteMsgsAndResponse(ctx, h, msgs, w, r, clog)
 }
 
+type v3OutboundPayload struct {
+	Messages []v3OutboundMessage `json:"messages"`
+}
+
+type v3OutboundMessage struct {
+	From         string                  `json:"from"`
+	Destinations []v3OutboundDestination `json:"destinations"`
+	Content      v3OutboundContent       `json:"content"`
+	Webhooks     *v3OutboundWebhooks     `json:"webhooks,omitempty"`
+}
+
+type v3OutboundDestination struct {
+	To        string `json:"to"`
+	MessageID string `json:"messageId,omitempty"`
+}
+
+type v3OutboundContent struct {
+	Text            string `json:"text,omitempty"`
+	Transliteration string `json:"transliteration,omitempty"`
+}
+
+type v3OutboundWebhooks struct {
+	Delivery v3OutboundDelivery `json:"delivery"`
+}
+
+type v3OutboundDelivery struct {
+	URL                string `json:"url"`
+	IntermediateReport bool   `json:"intermediateReport"`
+	ContentType        string `json:"contentType"`
+}
+
+type v2MMSOutboundPayload struct {
+	Messages []v2MMSOutboundMessage `json:"messages"`
+}
+
+type v2MMSOutboundMessage struct {
+	Sender       string                     `json:"sender"`
+	Destinations []v2MMSOutboundDestination `json:"destinations"`
+	Content      v2MMSOutboundContent       `json:"content"`
+	Webhooks     *v3OutboundWebhooks        `json:"webhooks,omitempty"` // Reusing SMS webhooks struct
+}
+
+type v2MMSOutboundDestination struct {
+	To        string `json:"to"`
+	MessageID string `json:"messageId,omitempty"`
+}
+
+type v2MMSOutboundContent struct {
+	Title           string                 `json:"title"`
+	MessageSegments []v2MMSMessageSegment  `json:"messageSegments"`
+}
+
+type v2MMSMessageSegment struct {
+	Type      string `json:"type"` // "TEXT" or "IMAGE", "AUDIO", "VIDEO"
+	ContentID string `json:"contentId,omitempty"`
+	Text      string `json:"text,omitempty"`
+	URL       string `json:"url,omitempty"` // For media content
+}
+
 func (h *handler) Send(ctx context.Context, msg courier.MsgOut, res *courier.SendResult, clog *courier.ChannelLog) error {
+	apiKey := msg.Channel().StringConfigForKey(models.ConfigAPIKey, "")
```

Comment:
Does the API V3 still allow basic auth?

We probably need the Basic auth only so that existing channels still work

Reply:
If that config is required, we need to update how the channel is connected too on the web UI